Detect TAP test prerequisites with CMake (#3299)
timescale/timescaledb: CMakeLists.txt

  add_link_options(--coverage)
endif (CODECOVERAGE)

# TAP test support
option(TAP "Enable TAP test support" ON)

if (TAP)

>   include(FindPerlLibs)
>   find_package(PerlLibs 5.8)

Reviewer: You don't need to include it first, and you don't really need to know where the libraries are, only where `perl` executable is.
```suggestion
  find_package(Perl 5.8)
```

Author: I use PerlLibs since it sets up a number of variables that are later used. Those are not set when looking for just Perl.

Reviewer: Which are those?

Author: https://cmake.org/cmake/help/latest/module/FindPerlLibs.html

See tsl/test/CMakeLists.txt

Author: Removed the need for extra paths, and thus PerlLibs check, since Nikhil confirmed it works for him without it.

  if (PERLLIBS_FOUND)
    get_filename_component(PERL_BIN_PATH ${PERL_EXECUTABLE} DIRECTORY)

>     find_program(PROVE prove
>       PATHS
>       ${PERL_BIN_PATH}
>       "/usr/bin"
>       NO_DEFAULT_PATH)

Reviewer: Not sure there is any difference in reality, but `HINTS` are recommended for "computed" paths (https://cmake.org/cmake/help/latest/command/find_program.html).

> Search the paths specified by the HINTS option. These should be paths computed by system introspection, such as a hint provided by the location of another item already found. Hard-coded guesses should be specified with the PATHS option.

It would be very strange if `perl` was installed but not `prove`, but better safe than sorry.

```suggestion
    find_program(PROVE prove
      HINTS ${PERL_BIN_PATH}
      PATHS "/usr/bin"
      NO_DEFAULT_PATH)
```

Author: It might still not be in the standard PATH, so finding the exact location is still worthwhile, especially when there are multiple location. We also want to use the "prove" matching the Perl we found.

Reviewer: Yes, hence the "better safe than sorry".

    if (NOT PROVE)
      message(FATAL_ERROR "The prove binary was not found. Disable TAP tests with -DTAP=OFF.")
    endif ()

>     #execute_process(COMMAND ${PERL_EXECUTABLE} -MIPC::Run -e ""
>     #  ERROR_QUIET RESULT_VARIABLE PERL_MODULE_STATUS)
> 
>     find_file(PERL_IPC_RUN Run.pm
>       PATHS
>       "${PERL_PRIVLIB}"
>       "${PERL_UPDATE_PRIVLIB}"
>       "${PERL_VENDORLIB}"
>       "${PERL_SITELIB}"
>       "/usr/share/perl"
>       "/usr/share/perl5"
>       "/usr/share/perl6"
>       "$ENV{HOME}/perl/lib/perl"
>       "$ENV{HOME}/perl5/lib/perl5"
>       "$ENV{HOME}/perl6/lib/perl6"
>       PATH_SUFFIXES
>       IPC)
> 
>     if (NOT PERL_IPC_RUN)
>       message(FATAL_ERROR "IPC::Run Perl module not found. Disable TAP tests with -DTAP=OFF.")
>     endif ()

Reviewer: If there is a file `Run.pm` in the top directories, it will find that first. Using `PATH_SUFFIXES` just provides *extra* directories to search. I would suggest that you use the method you commented out instead to check that the library is installed:
```suggestion
execute_process(
    COMMAND perl -mIPC::Run -e ''
    RESULT_VARIABLE status
    ERROR_QUIET)
    # If status is non-zero (true), module does not exist.
    if (status)
      message(FATAL_ERROR "IPC::Run Perl module not found. Disable TAP tests with -DTAP=OFF.")
    endif ()
```
